src/odemis/driver/andorshrk.py

```diff
@@ -623,6 +654,54 @@ def Initialize(self):
     def Close(self):
         self._dll.ShamrockClose()
 
+    def _reconnect(self):
+        """
+        Attempt to reconnect the device. It will block until this happens.
+        On return, the hardware should be ready to use as before, excepted it
+        still needs the settings to be applied.
+        """
+        self.state._set_value(HwError("Shamrock disconnected"), force_write=True)
+        self._reconnecting = True
+        logging.debug("Reconnecting Shamrock...")
+        self.Close()
+
+        # In order to make reestablish the connection, we need to turn the power off and on again and then
+        # reinitialize.
+        logging.debug("Cycling power...")
+        if self._power_supplier:
```

Comment:
@pieleric Do you think this is a good idea to implement power cycling? Maybe we can also to it for the pmtctrl and tmcm instead of uhubctl. For this driver, it seems necessary to do this.

Reply:
The main problem with the power cycling is that it takes so much time (90 s in the bad cases). So it would be good to find a faster alternative. Did you check if using the uhubctl it is possible to find the device again? If so, it would be worth implementing. There are other types of USB disconnection in Linux (eg via /sys/), they might also be worth checking too as might be easier to implement.
I think that you could also have several of the approaches implented, and they would be use as fallback for one another.

Reply:
Even if we do it with `uhubctl`, we need to reinitialize the spectrograph otherwise we get the 'device not initialized' error back from the hardware. This takes quite a long time anyway.

Reply:
Every device which as a power_supplier has a `powerSupply` VA, which can be set False/True. => Use that VA.